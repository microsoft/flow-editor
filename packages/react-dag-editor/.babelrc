{
  "presets": [
    [
      "@nrwl/react/babel",
      {
        "runtime": "automatic",
        "useBuiltIns": false
      }
    ]
  ],
  "plugins": [
    [
      "@babel/plugin-transform-typescript",
      {
        "allowDeclareFields": true
      }
    ]
  ]
}
