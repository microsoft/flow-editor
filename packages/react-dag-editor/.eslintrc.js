module.exports = {
  extends: ["@react-dag-editor-common/eslint-config"]
};
