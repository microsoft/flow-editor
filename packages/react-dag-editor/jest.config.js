module.exports = {
  displayName: 'react-dag-editor',
  preset: '../../jest.preset.js',
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/packages/react-dag-editor',
};
