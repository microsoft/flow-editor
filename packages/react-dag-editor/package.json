{
  "name": "react-dag-editor",
  "license": "MIT",
  "repository": {
    "url": "https://github.com/microsoft/react-dag-editor.git"
  },
  "version": "0.4.2",
  "dependencies": {
    "@fluentui/merge-styles": "^8.2.0",
    "eventemitter3": "^4.0.7",
    "react-jss": "~10.2.0",
    "toposort": "^1.0.7",
    "uuid": "^7.0.3"
  }
}
