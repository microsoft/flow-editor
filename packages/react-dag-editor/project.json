{
  "name": "react-dag-editor",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "packages/react-dag-editor/src",
  "projectType": "library",
  "targets": {
    "build": {
      "executor": "@nrwl/rollup:rollup",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/packages/react-dag-editor",
        "tsConfig": "packages/react-dag-editor/tsconfig.lib.json",
        "project": "packages/react-dag-editor/package.json",
        "entryFile": "packages/react-dag-editor/src/index.ts",
        "external": ["react/jsx-runtime"],
        "rollupConfig": "packages/react-dag-editor/rollup.config.js",
        "extractCss": false,
        "assets": [
          {
            "glob": "README.md",
            "input": ".",
            "output": "."
          },
          {
            "glob": "packages/react-dag-editor/src",
            "input": ".",
            "output": "."
          }
        ]
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["packages/react-dag-editor/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/packages/react-dag-editor"],
      "options": {
        "jestConfig": "packages/react-dag-editor/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "storybook": {
      "executor": "@nrwl/storybook:storybook",
      "options": {
        "uiFramework": "@storybook/react",
        "port": 4400,
        "configDir": "packages/react-dag-editor/.storybook"
      },
      "configurations": {
        "ci": {
          "quiet": true
        }
      }
    },
    "build-storybook": {
      "executor": "@nrwl/storybook:build",
      "outputs": ["{options.outputDir}"],
      "options": {
        "uiFramework": "@storybook/react",
        "configDir": "packages/react-dag-editor/.storybook",
        "outputDir": "dist/storybook/react-dag-editor"
      },
      "configurations": {
        "ci": {
          "quiet": true
        }
      }
    },
    "typedoc": {
      "executor": "nx:run-commands",
      "options": {
        "commands": [
          "yarn typedoc --tsconfig packages/react-dag-editor/tsconfig.lib.json --out dist/storybook/react-dag-editor/apidoc packages/react-dag-editor/src/index.ts"
        ]
      }
    },
    "site": {
      "executor": "nx:run-commands",
      "options": {
        "parallel": false,
        "commands": [
          "yarn nx run react-dag-editor:build-storybook",
          "yarn nx run react-dag-editor:typedoc"
        ]
      }
    }
  }
}
