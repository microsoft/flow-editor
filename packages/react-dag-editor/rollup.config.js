const path = require("path");
const deepmerge = require("deepmerge");
const { commonConfig, outputConfig } = require("@react-dag-editor-common/builder");

const mainConfig = deepmerge(
  {
    input: {
      index: "./src/index.ts"
    },
    output: outputConfig("ReactFlowEditorEngine", process.cwd())
  },
  commonConfig(process.cwd())
);

const vendorConfig = deepmerge(
  {
    input: {
      vendor: "./src/libs/index.ts"
    },
    output: {
      format: "amd",
      sourcemap: true,
      dir: path.resolve(process.cwd(), "dist"),
      entryFileNames: "[name].js"
    }
  },
  commonConfig(process.cwd())
);

const config = [mainConfig, vendorConfig];

module.exports = config;
