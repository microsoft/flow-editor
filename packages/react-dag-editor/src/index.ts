export * from "./lib/components";
export * from "./lib/contexts";
export * from "./lib/built-in";
export * from "./lib/utils";
export * from "./lib/hooks";
export * from "./lib/testAPI";
export * from "./lib/Features";
export * from "./lib/reducers/composeReducers";
export * from "./lib/models/status";
export * from "./lib/models/port";
export * from "./lib/models/node";
export * from "./lib/models/edge";
export * from "./lib/models/canvas";
export * from "./lib/models/ContentState";
export * from "./lib/models/geometry";
export * from "./lib/models/state";
export * from "./lib/models/event";
export * from "./lib/models/config/GraphConfig";
export * from "./lib/models/config/GraphConfigBuilder";
export * from "./lib/models/config/types";
export * from "./lib/createGraphState";

export * as ContentUtils from "./lib/content-utils";
export * as NodeUtils from "./lib/node-utils";
