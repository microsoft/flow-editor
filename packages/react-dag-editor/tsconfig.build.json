{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "composite": false,
    "rootDir": "./src"
  },
  "include": [
    "src/**/*"
  ],
  "exclude": [
    "node_modules",
    "dist",
    "scripts"
  ],
  "typeRoots": [
    "./node_modules/@types",
    "./src/@types/"
  ]
}
