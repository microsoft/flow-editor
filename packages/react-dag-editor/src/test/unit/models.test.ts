import { GraphModel } from "react-dag-editor";
import { getSample1Data } from "./__data__/getSample1Data";

it("should convert from plain objects", () => {
  const graph = GraphModel.fromJSON(getSample1Data());
  expect(graph.nodes).toMatchSnapshot();
  expect(graph.edges).toMatchSnapshot();
  expect(graph.head).toMatchSnapshot();
  expect(graph.tail).toMatchSnapshot();
  expect(graph.edgesBySource).toMatchSnapshot();
  expect(graph.edgesByTarget).toMatchSnapshot();
});

it("should convert to plain objects", () => {
  expect(GraphModel.fromJSON(getSample1Data()).toJSON()).toMatchSnapshot();
});
