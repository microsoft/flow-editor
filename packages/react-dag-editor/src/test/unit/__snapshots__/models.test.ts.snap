// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should convert from plain objects 1`] = `
OrderedMap {
  "hashRoot": BitmapIndexedNode {
    "children": Array [],
    "dataMap": 754729,
    "hashes": Array [
      347532448,
      1062478147,
      978270693,
      592768554,
      484045359,
      321069040,
      817570257,
      1014460051,
    ],
    "keys": Array [
      "47566002",
      "fb404f70",
      "81777c66",
      "1fd27f80",
      "c4ee5025",
      "d1647408",
      "4b199015",
      "f6dc0eb1",
    ],
    "nodeMap": 0,
    "owner": 7,
    "size": 8,
    "type": 0,
    "values": Array [
      1,
      5,
      7,
      8,
      6,
      3,
      4,
      2,
    ],
  },
  "id": 8,
  "itemId": 8,
  "sortedRoot": LeafNode {
    "keys": Array [
      1,
      2,
      3,
      4,
      5,
      6,
      7,
      8,
    ],
    "owner": 7,
    "type": 1,
    "values": Array [
      Array [
        "47566002",
        Object {
          "ariaLabel": "test aria label for node Select Columns in Dataset",
          "automationId": undefined,
          "data": Object {
            "comment": "",
            "iconName": "TestUserSolid",
            "status": "completed",
            "statusIcon": "completed",
          },
          "height": undefined,
          "id": "47566002",
          "name": "Select Columns in Dataset",
          "ports": Array [
            Object {
              "ariaLabel": "test aria label for port Dataset",
              "id": "port-0",
              "isInputDisabled": false,
              "isOutputDisabled": true,
              "name": "Dataset",
              "position": Array [
                0.5,
                0,
              ],
              "shape": "default",
              "status": 0,
            },
            Object {
              "ariaLabel": "test aria label for port Results_dataset",
              "id": "port-1",
              "isInputDisabled": true,
              "isOutputDisabled": false,
              "name": "Results_dataset",
              "position": Array [
                0.5,
                1,
              ],
              "shape": "default",
              "status": 0,
            },
          ],
          "shape": undefined,
          "status": 0,
          "width": undefined,
          "x": 548,
          "y": 162,
        },
      ],
      Array [
        "f6dc0eb1",
        Object {
          "ariaLabel": "test aria label for node Clean Missing Data",
          "automationId": undefined,
          "data": Object {
            "comment": "",
            "iconName": "TestUserSolid",
            "status": "completed",
            "statusIcon": "completed",
          },
          "height": undefined,
          "id": "f6dc0eb1",
          "name": "Clean Missing Data",
          "ports": Array [
            Object {
              "ariaLabel": "test aria label for port Dataset",
              "id": "port-0",
              "isInputDisabled": false,
              "isOutputDisabled": true,
              "name": "Dataset",
              "position": Array [
                0.5,
                0,
              ],
              "shape": "default",
              "status": 0,
            },
            Object {
              "ariaLabel": "test aria label for port Cleaned_dataset",
              "id": "port-1",
              "isInputDisabled": true,
              "isOutputDisabled": false,
              "name": "Cleaned_dataset",
              "position": Array [
                0.3333333333333333,
                1,
              ],
              "shape": "default",
              "status": 0,
            },
            Object {
              "ariaLabel": "test aria label for port Cleaning_transformation",
              "id": "port-2",
              "isInputDisabled": true,
              "isOutputDisabled": false,
              "name": "Cleaning_transformation",
              "position": Array [
                0.6666666666666666,
                1,
              ],
              "shape": "default",
              "status": 0,
            },
          ],
          "shape": undefined,
          "status": 0,
          "width": undefined,
          "x": 548,
          "y": 280,
        },
      ],
      Array [
        "d1647408",
        Object {
          "ariaLabel": "test aria label for node Split Data",
          "automationId": undefined,
          "data": Object {
            "comment": "",
            "iconName": "TestUserSolid",
            "status": "completed",
            "statusIcon": "completed",
          },
          "height": undefined,
          "id": "d1647408",
          "name": "Split Data",
          "ports": Array [
            Object {
              "ariaLabel": "test aria label for port Dataset",
              "id": "port-0",
              "isInputDisabled": false,
              "isOutputDisabled": true,
              "name": "Dataset",
              "position": Array [
                0.5,
                0,
              ],
              "shape": "default",
              "status": 0,
            },
            Object {
              "ariaLabel": "test aria label for port Results_dataset1",
              "id": "port-1",
              "isInputDisabled": true,
              "isOutputDisabled": false,
              "name": "Results_dataset1",
              "position": Array [
                0.3333333333333333,
                1,
              ],
              "shape": "default",
              "status": 0,
            },
            Object {
              "ariaLabel": "test aria label for port Results_dataset2",
              "id": "port-2",
              "isInputDisabled": true,
              "isOutputDisabled": false,
              "name": "Results_dataset2",
              "position": Array [
                0.6666666666666666,
                1,
              ],
              "shape": "default",
              "status": 0,
            },
          ],
          "shape": undefined,
          "status": 0,
          "width": undefined,
          "x": 542,
          "y": 386,
        },
      ],
      Array [
        "4b199015",
        Object {
          "ariaLabel": "test aria label for node Linear Regression",
          "automationId": undefined,
          "data": Object {
            "comment": "",
            "iconName": "MachineLearning",
            "status": "completed",
            "statusIcon": "completed",
          },
          "height": undefined,
          "id": "4b199015",
          "name": "Linear Regression",
          "ports": Array [
            Object {
              "ariaLabel": "test aria label for port Untrained_model",
              "id": "port-0",
              "isInputDisabled": true,
              "isOutputDisabled": false,
              "name": "Untrained_model",
              "position": Array [
                0.5,
                1,
              ],
              "shape": "default",
              "status": 0,
            },
          ],
          "shape": undefined,
          "status": 0,
          "width": undefined,
          "x": 173,
          "y": 393.5,
        },
      ],
      Array [
        "fb404f70",
        Object {
          "ariaLabel": "test aria label for node Train Model",
          "automationId": undefined,
          "data": Object {
            "comment": "",
            "iconName": "ModelingView",
            "status": "running",
            "statusIcon": "running",
          },
          "height": undefined,
          "id": "fb404f70",
          "name": "Train Model",
          "ports": Array [
            Object {
              "ariaLabel": "test aria label for port Untrained_model",
              "id": "port-0",
              "isInputDisabled": false,
              "isOutputDisabled": true,
              "name": "Untrained_model",
              "position": Array [
                0.3333333333333333,
                0,
              ],
              "shape": "default",
              "status": 0,
            },
            Object {
              "ariaLabel": "test aria label for port Dataset",
              "id": "port-1",
              "isInputDisabled": false,
              "isOutputDisabled": true,
              "name": "Dataset",
              "position": Array [
                0.6666666666666666,
                0,
              ],
              "shape": "default",
              "status": 0,
            },
            Object {
              "ariaLabel": "test aria label for port Trained_model",
              "id": "port-2",
              "isInputDisabled": true,
              "isOutputDisabled": false,
              "name": "Trained_model",
              "position": Array [
                0.5,
                1,
              ],
              "shape": "default",
              "status": 0,
            },
          ],
          "shape": undefined,
          "status": 0,
          "width": undefined,
          "x": 387,
          "y": 517,
        },
      ],
      Array [
        "c4ee5025",
        Object {
          "ariaLabel": "test aria label for node Score Model",
          "automationId": undefined,
          "data": Object {
            "comment": "",
            "iconName": "WebAppBuilderModule",
            "status": "running",
            "statusIcon": "running",
          },
          "height": undefined,
          "id": "c4ee5025",
          "name": "Score Model",
          "ports": Array [
            Object {
              "ariaLabel": "test aria label for port Trained_model",
              "id": "port-0",
              "isInputDisabled": false,
              "isOutputDisabled": true,
              "name": "Trained_model",
              "position": Array [
                0.3333333333333333,
                0,
              ],
              "shape": "default",
              "status": 0,
            },
            Object {
              "ariaLabel": "test aria label for port Dataset",
              "id": "port-1",
              "isInputDisabled": false,
              "isOutputDisabled": true,
              "name": "Dataset",
              "position": Array [
                0.6666666666666666,
                0,
              ],
              "shape": "default",
              "status": 0,
            },
            Object {
              "ariaLabel": "test aria label for port Scored_dataset",
              "id": "port-2",
              "isInputDisabled": true,
              "isOutputDisabled": false,
              "name": "Scored_dataset",
              "position": Array [
                0.5,
                1,
              ],
              "shape": "default",
              "status": 0,
            },
          ],
          "shape": undefined,
          "status": 0,
          "width": undefined,
          "x": 491,
          "y": 630,
        },
      ],
      Array [
        "81777c66",
        Object {
          "ariaLabel": "test aria label for node Evaluate Model",
          "automationId": undefined,
          "data": Object {
            "comment": "",
            "iconName": "WebAppBuilderModule",
            "status": "running",
            "statusIcon": "running",
          },
          "height": undefined,
          "id": "81777c66",
          "name": "Evaluate Model",
          "ports": Array [
            Object {
              "ariaLabel": "test aria label for port Scored_dataset",
              "id": "port-0",
              "isInputDisabled": false,
              "isOutputDisabled": true,
              "name": "Scored_dataset",
              "position": Array [
                0.3333333333333333,
                0,
              ],
              "shape": "default",
              "status": 0,
            },
            Object {
              "ariaLabel": "test aria label for port Scored_dataset_to_compare",
              "id": "port-1",
              "isInputDisabled": false,
              "isOutputDisabled": true,
              "name": "Scored_dataset_to_compare",
              "position": Array [
                0.6666666666666666,
                0,
              ],
              "shape": "default",
              "status": 0,
            },
            Object {
              "ariaLabel": "test aria label for port Evaluation_results",
              "id": "port-2",
              "isInputDisabled": true,
              "isOutputDisabled": false,
              "name": "Evaluation_results",
              "position": Array [
                0.5,
                1,
              ],
              "shape": "default",
              "status": 0,
            },
          ],
          "shape": undefined,
          "status": 0,
          "width": undefined,
          "x": 493,
          "y": 738,
        },
      ],
      Array [
        "1fd27f80",
        Object {
          "ariaLabel": "test aria label for node Automobile price data (Raw)",
          "automationId": undefined,
          "data": Object {
            "comment": "",
            "iconName": "Database",
            "status": "running",
            "statusIcon": "running",
          },
          "height": undefined,
          "id": "1fd27f80",
          "name": "Automobile price data (Raw)",
          "ports": Array [
            Object {
              "ariaLabel": "test aria label for port DatasetOutput",
              "id": "port-0",
              "isInputDisabled": true,
              "isOutputDisabled": false,
              "name": "DatasetOutput",
              "position": Array [
                0.5,
                1,
              ],
              "shape": "default",
              "status": 0,
            },
          ],
          "shape": undefined,
          "status": 0,
          "width": undefined,
          "x": 544,
          "y": 49,
        },
      ],
    ],
  },
}
`;

exports[`should convert from plain objects 2`] = `
HashMap {
  "id": 18,
  "root": BitmapIndexedNode {
    "children": Array [],
    "dataMap": 16711680,
    "hashes": Array [
      48,
      49,
      50,
      51,
      52,
      53,
      54,
      55,
    ],
    "keys": Array [
      "0",
      "1",
      "2",
      "3",
      "4",
      "5",
      "6",
      "7",
    ],
    "nodeMap": 0,
    "owner": 15,
    "size": 8,
    "type": 0,
    "values": Array [
      Object {
        "automationId": undefined,
        "data": undefined,
        "id": "0",
        "shape": undefined,
        "source": "d1647408",
        "sourcePortId": "port-1",
        "status": 0,
        "target": "fb404f70",
        "targetPortId": "port-1",
      },
      Object {
        "automationId": undefined,
        "data": undefined,
        "id": "1",
        "shape": undefined,
        "source": "47566002",
        "sourcePortId": "port-1",
        "status": 0,
        "target": "f6dc0eb1",
        "targetPortId": "port-0",
      },
      Object {
        "automationId": undefined,
        "data": undefined,
        "id": "2",
        "shape": undefined,
        "source": "f6dc0eb1",
        "sourcePortId": "port-1",
        "status": 0,
        "target": "d1647408",
        "targetPortId": "port-0",
      },
      Object {
        "automationId": undefined,
        "data": undefined,
        "id": "3",
        "shape": undefined,
        "source": "4b199015",
        "sourcePortId": "port-0",
        "status": 0,
        "target": "fb404f70",
        "targetPortId": "port-0",
      },
      Object {
        "automationId": undefined,
        "data": undefined,
        "id": "4",
        "shape": undefined,
        "source": "fb404f70",
        "sourcePortId": "port-2",
        "status": 0,
        "target": "c4ee5025",
        "targetPortId": "port-0",
      },
      Object {
        "automationId": undefined,
        "data": undefined,
        "id": "5",
        "shape": undefined,
        "source": "d1647408",
        "sourcePortId": "port-2",
        "status": 0,
        "target": "c4ee5025",
        "targetPortId": "port-1",
      },
      Object {
        "automationId": undefined,
        "data": undefined,
        "id": "6",
        "shape": undefined,
        "source": "c4ee5025",
        "sourcePortId": "port-2",
        "status": 0,
        "target": "81777c66",
        "targetPortId": "port-0",
      },
      Object {
        "automationId": undefined,
        "data": undefined,
        "id": "7",
        "shape": undefined,
        "source": "1fd27f80",
        "sourcePortId": "port-0",
        "status": 0,
        "target": "47566002",
        "targetPortId": "port-0",
      },
    ],
  },
}
`;

exports[`should convert from plain objects 3`] = `"47566002"`;

exports[`should convert from plain objects 4`] = `"1fd27f80"`;

exports[`should convert from plain objects 5`] = `
HashMap {
  "id": 19,
  "root": BitmapIndexedNode {
    "children": Array [],
    "dataMap": 754697,
    "hashes": Array [
      347532448,
      1062478147,
      592768554,
      484045359,
      321069040,
      817570257,
      1014460051,
    ],
    "keys": Array [
      "47566002",
      "fb404f70",
      "1fd27f80",
      "c4ee5025",
      "d1647408",
      "4b199015",
      "f6dc0eb1",
    ],
    "nodeMap": 0,
    "owner": 16,
    "size": 7,
    "type": 0,
    "values": Array [
      Map {
        "port-1" => Set {
          "1",
        },
      },
      Map {
        "port-2" => Set {
          "4",
        },
      },
      Map {
        "port-0" => Set {
          "7",
        },
      },
      Map {
        "port-2" => Set {
          "6",
        },
      },
      Map {
        "port-1" => Set {
          "0",
        },
        "port-2" => Set {
          "5",
        },
      },
      Map {
        "port-0" => Set {
          "3",
        },
      },
      Map {
        "port-1" => Set {
          "2",
        },
      },
    ],
  },
}
`;

exports[`should convert from plain objects 6`] = `
HashMap {
  "id": 20,
  "root": BitmapIndexedNode {
    "children": Array [],
    "dataMap": 622633,
    "hashes": Array [
      347532448,
      1062478147,
      978270693,
      484045359,
      321069040,
      1014460051,
    ],
    "keys": Array [
      "47566002",
      "fb404f70",
      "81777c66",
      "c4ee5025",
      "d1647408",
      "f6dc0eb1",
    ],
    "nodeMap": 0,
    "owner": 17,
    "size": 6,
    "type": 0,
    "values": Array [
      Map {
        "port-0" => Set {
          "7",
        },
      },
      Map {
        "port-1" => Set {
          "0",
        },
        "port-0" => Set {
          "3",
        },
      },
      Map {
        "port-0" => Set {
          "6",
        },
      },
      Map {
        "port-0" => Set {
          "4",
        },
        "port-1" => Set {
          "5",
        },
      },
      Map {
        "port-0" => Set {
          "2",
        },
      },
      Map {
        "port-0" => Set {
          "1",
        },
      },
    ],
  },
}
`;

exports[`should convert to plain objects 1`] = `
Object {
  "edges": Array [
    Object {
      "automationId": undefined,
      "data": undefined,
      "id": "0",
      "shape": undefined,
      "source": "d1647408",
      "sourcePortId": "port-1",
      "status": 0,
      "target": "fb404f70",
      "targetPortId": "port-1",
    },
    Object {
      "automationId": undefined,
      "data": undefined,
      "id": "1",
      "shape": undefined,
      "source": "47566002",
      "sourcePortId": "port-1",
      "status": 0,
      "target": "f6dc0eb1",
      "targetPortId": "port-0",
    },
    Object {
      "automationId": undefined,
      "data": undefined,
      "id": "2",
      "shape": undefined,
      "source": "f6dc0eb1",
      "sourcePortId": "port-1",
      "status": 0,
      "target": "d1647408",
      "targetPortId": "port-0",
    },
    Object {
      "automationId": undefined,
      "data": undefined,
      "id": "3",
      "shape": undefined,
      "source": "4b199015",
      "sourcePortId": "port-0",
      "status": 0,
      "target": "fb404f70",
      "targetPortId": "port-0",
    },
    Object {
      "automationId": undefined,
      "data": undefined,
      "id": "4",
      "shape": undefined,
      "source": "fb404f70",
      "sourcePortId": "port-2",
      "status": 0,
      "target": "c4ee5025",
      "targetPortId": "port-0",
    },
    Object {
      "automationId": undefined,
      "data": undefined,
      "id": "5",
      "shape": undefined,
      "source": "d1647408",
      "sourcePortId": "port-2",
      "status": 0,
      "target": "c4ee5025",
      "targetPortId": "port-1",
    },
    Object {
      "automationId": undefined,
      "data": undefined,
      "id": "6",
      "shape": undefined,
      "source": "c4ee5025",
      "sourcePortId": "port-2",
      "status": 0,
      "target": "81777c66",
      "targetPortId": "port-0",
    },
    Object {
      "automationId": undefined,
      "data": undefined,
      "id": "7",
      "shape": undefined,
      "source": "1fd27f80",
      "sourcePortId": "port-0",
      "status": 0,
      "target": "47566002",
      "targetPortId": "port-0",
    },
  ],
  "nodes": Array [
    Object {
      "ariaLabel": "test aria label for node Select Columns in Dataset",
      "automationId": undefined,
      "data": Object {
        "comment": "",
        "iconName": "TestUserSolid",
        "status": "completed",
        "statusIcon": "completed",
      },
      "height": undefined,
      "id": "47566002",
      "name": "Select Columns in Dataset",
      "ports": Array [
        Object {
          "ariaLabel": "test aria label for port Dataset",
          "id": "port-0",
          "isInputDisabled": false,
          "isOutputDisabled": true,
          "name": "Dataset",
          "position": Array [
            0.5,
            0,
          ],
          "shape": "default",
          "status": 0,
        },
        Object {
          "ariaLabel": "test aria label for port Results_dataset",
          "id": "port-1",
          "isInputDisabled": true,
          "isOutputDisabled": false,
          "name": "Results_dataset",
          "position": Array [
            0.5,
            1,
          ],
          "shape": "default",
          "status": 0,
        },
      ],
      "shape": undefined,
      "status": 0,
      "width": undefined,
      "x": 548,
      "y": 162,
    },
    Object {
      "ariaLabel": "test aria label for node Clean Missing Data",
      "automationId": undefined,
      "data": Object {
        "comment": "",
        "iconName": "TestUserSolid",
        "status": "completed",
        "statusIcon": "completed",
      },
      "height": undefined,
      "id": "f6dc0eb1",
      "name": "Clean Missing Data",
      "ports": Array [
        Object {
          "ariaLabel": "test aria label for port Dataset",
          "id": "port-0",
          "isInputDisabled": false,
          "isOutputDisabled": true,
          "name": "Dataset",
          "position": Array [
            0.5,
            0,
          ],
          "shape": "default",
          "status": 0,
        },
        Object {
          "ariaLabel": "test aria label for port Cleaned_dataset",
          "id": "port-1",
          "isInputDisabled": true,
          "isOutputDisabled": false,
          "name": "Cleaned_dataset",
          "position": Array [
            0.3333333333333333,
            1,
          ],
          "shape": "default",
          "status": 0,
        },
        Object {
          "ariaLabel": "test aria label for port Cleaning_transformation",
          "id": "port-2",
          "isInputDisabled": true,
          "isOutputDisabled": false,
          "name": "Cleaning_transformation",
          "position": Array [
            0.6666666666666666,
            1,
          ],
          "shape": "default",
          "status": 0,
        },
      ],
      "shape": undefined,
      "status": 0,
      "width": undefined,
      "x": 548,
      "y": 280,
    },
    Object {
      "ariaLabel": "test aria label for node Split Data",
      "automationId": undefined,
      "data": Object {
        "comment": "",
        "iconName": "TestUserSolid",
        "status": "completed",
        "statusIcon": "completed",
      },
      "height": undefined,
      "id": "d1647408",
      "name": "Split Data",
      "ports": Array [
        Object {
          "ariaLabel": "test aria label for port Dataset",
          "id": "port-0",
          "isInputDisabled": false,
          "isOutputDisabled": true,
          "name": "Dataset",
          "position": Array [
            0.5,
            0,
          ],
          "shape": "default",
          "status": 0,
        },
        Object {
          "ariaLabel": "test aria label for port Results_dataset1",
          "id": "port-1",
          "isInputDisabled": true,
          "isOutputDisabled": false,
          "name": "Results_dataset1",
          "position": Array [
            0.3333333333333333,
            1,
          ],
          "shape": "default",
          "status": 0,
        },
        Object {
          "ariaLabel": "test aria label for port Results_dataset2",
          "id": "port-2",
          "isInputDisabled": true,
          "isOutputDisabled": false,
          "name": "Results_dataset2",
          "position": Array [
            0.6666666666666666,
            1,
          ],
          "shape": "default",
          "status": 0,
        },
      ],
      "shape": undefined,
      "status": 0,
      "width": undefined,
      "x": 542,
      "y": 386,
    },
    Object {
      "ariaLabel": "test aria label for node Linear Regression",
      "automationId": undefined,
      "data": Object {
        "comment": "",
        "iconName": "MachineLearning",
        "status": "completed",
        "statusIcon": "completed",
      },
      "height": undefined,
      "id": "4b199015",
      "name": "Linear Regression",
      "ports": Array [
        Object {
          "ariaLabel": "test aria label for port Untrained_model",
          "id": "port-0",
          "isInputDisabled": true,
          "isOutputDisabled": false,
          "name": "Untrained_model",
          "position": Array [
            0.5,
            1,
          ],
          "shape": "default",
          "status": 0,
        },
      ],
      "shape": undefined,
      "status": 0,
      "width": undefined,
      "x": 173,
      "y": 393.5,
    },
    Object {
      "ariaLabel": "test aria label for node Train Model",
      "automationId": undefined,
      "data": Object {
        "comment": "",
        "iconName": "ModelingView",
        "status": "running",
        "statusIcon": "running",
      },
      "height": undefined,
      "id": "fb404f70",
      "name": "Train Model",
      "ports": Array [
        Object {
          "ariaLabel": "test aria label for port Untrained_model",
          "id": "port-0",
          "isInputDisabled": false,
          "isOutputDisabled": true,
          "name": "Untrained_model",
          "position": Array [
            0.3333333333333333,
            0,
          ],
          "shape": "default",
          "status": 0,
        },
        Object {
          "ariaLabel": "test aria label for port Dataset",
          "id": "port-1",
          "isInputDisabled": false,
          "isOutputDisabled": true,
          "name": "Dataset",
          "position": Array [
            0.6666666666666666,
            0,
          ],
          "shape": "default",
          "status": 0,
        },
        Object {
          "ariaLabel": "test aria label for port Trained_model",
          "id": "port-2",
          "isInputDisabled": true,
          "isOutputDisabled": false,
          "name": "Trained_model",
          "position": Array [
            0.5,
            1,
          ],
          "shape": "default",
          "status": 0,
        },
      ],
      "shape": undefined,
      "status": 0,
      "width": undefined,
      "x": 387,
      "y": 517,
    },
    Object {
      "ariaLabel": "test aria label for node Score Model",
      "automationId": undefined,
      "data": Object {
        "comment": "",
        "iconName": "WebAppBuilderModule",
        "status": "running",
        "statusIcon": "running",
      },
      "height": undefined,
      "id": "c4ee5025",
      "name": "Score Model",
      "ports": Array [
        Object {
          "ariaLabel": "test aria label for port Trained_model",
          "id": "port-0",
          "isInputDisabled": false,
          "isOutputDisabled": true,
          "name": "Trained_model",
          "position": Array [
            0.3333333333333333,
            0,
          ],
          "shape": "default",
          "status": 0,
        },
        Object {
          "ariaLabel": "test aria label for port Dataset",
          "id": "port-1",
          "isInputDisabled": false,
          "isOutputDisabled": true,
          "name": "Dataset",
          "position": Array [
            0.6666666666666666,
            0,
          ],
          "shape": "default",
          "status": 0,
        },
        Object {
          "ariaLabel": "test aria label for port Scored_dataset",
          "id": "port-2",
          "isInputDisabled": true,
          "isOutputDisabled": false,
          "name": "Scored_dataset",
          "position": Array [
            0.5,
            1,
          ],
          "shape": "default",
          "status": 0,
        },
      ],
      "shape": undefined,
      "status": 0,
      "width": undefined,
      "x": 491,
      "y": 630,
    },
    Object {
      "ariaLabel": "test aria label for node Evaluate Model",
      "automationId": undefined,
      "data": Object {
        "comment": "",
        "iconName": "WebAppBuilderModule",
        "status": "running",
        "statusIcon": "running",
      },
      "height": undefined,
      "id": "81777c66",
      "name": "Evaluate Model",
      "ports": Array [
        Object {
          "ariaLabel": "test aria label for port Scored_dataset",
          "id": "port-0",
          "isInputDisabled": false,
          "isOutputDisabled": true,
          "name": "Scored_dataset",
          "position": Array [
            0.3333333333333333,
            0,
          ],
          "shape": "default",
          "status": 0,
        },
        Object {
          "ariaLabel": "test aria label for port Scored_dataset_to_compare",
          "id": "port-1",
          "isInputDisabled": false,
          "isOutputDisabled": true,
          "name": "Scored_dataset_to_compare",
          "position": Array [
            0.6666666666666666,
            0,
          ],
          "shape": "default",
          "status": 0,
        },
        Object {
          "ariaLabel": "test aria label for port Evaluation_results",
          "id": "port-2",
          "isInputDisabled": true,
          "isOutputDisabled": false,
          "name": "Evaluation_results",
          "position": Array [
            0.5,
            1,
          ],
          "shape": "default",
          "status": 0,
        },
      ],
      "shape": undefined,
      "status": 0,
      "width": undefined,
      "x": 493,
      "y": 738,
    },
    Object {
      "ariaLabel": "test aria label for node Automobile price data (Raw)",
      "automationId": undefined,
      "data": Object {
        "comment": "",
        "iconName": "Database",
        "status": "running",
        "statusIcon": "running",
      },
      "height": undefined,
      "id": "1fd27f80",
      "name": "Automobile price data (Raw)",
      "ports": Array [
        Object {
          "ariaLabel": "test aria label for port DatasetOutput",
          "id": "port-0",
          "isInputDisabled": true,
          "isOutputDisabled": false,
          "name": "DatasetOutput",
          "position": Array [
            0.5,
            1,
          ],
          "shape": "default",
          "status": 0,
        },
      ],
      "shape": undefined,
      "status": 0,
      "width": undefined,
      "x": 544,
      "y": 49,
    },
  ],
}
`;
